src/no_std/node.rs

```diff
@@ -62,14 +69,39 @@ pub(crate) enum Node<T> {
     Waiting(Task),
 }
 
+pub(crate) struct WaitingListener(Arc<TaskWaiting>);
+
+impl<T> fmt::Debug for Node<T> {
```

Comment:
why can't we just implement this on `TaskWaiting` instead?

Reply:
This struct was an artifact from when I was trying to track down where the `TaskWaiting` was going. I've now removed it.